workspaces/ui/src/components/diff/v2/Finalize.js

```diff
@@ -76,7 +71,18 @@ export default function FinalizeDialog(props) {
       JsonHelper.jsArrayToVector([EndBatchCommit(batchId)])
     );
     await specService.saveEvents(newEventStore, rfcId);
-    setTimeout(() => history.push(`${baseUrl}/diffs/${captureId}`), 500);
+    
+    setTimeout(() => {
+      history.push(`${baseUrl}/diffs/${captureId}`)
+    }, 500);
+
+    setTimeout(() => {
+      track('Committed Changes to Endpoint', {
+        message: commitMessage,
+        captureId,
+        suggestions: acceptedSuggestions.length,
+      });
+    }, 1000)
```

Comment:
What's the goal with the delay here as well?

Reply:
The timeout here exists really just so it comes after the history push finishes from above — otherwise another event occurs and takes priority, making it so the postcommit message doesn't show up

Reply:
haven't found a way to do this without the delay here 😕